### PR TITLE
BATCH-2462 Allow JpaItemWriter to support persist rather than merge for performance.

### PR DESCRIPTION
**Pull request by Naros**
I have signed and agree to the terms of the SpringSource Individual
Contributor License Agreement.

### CODE REVIEW

File: spring-batch-infrastructure/src/test/java/org/springframework/batch/item/database/JpaItemWriterPersistTests.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	}
+	
+	@Test
+	public void testPersist() throws Exception {
+		EntityManager em = mock(EntityManager.class, "em");
+		TransactionSynchronizationManager.bindResource(emf, new EntityManagerHolder(em));
+		List<String> items = Arrays.asList(new String[] { "persist1", "persist2" });
+		writer.write(items);		
+		TransactionSynchronizationManager.unbindResource(emf);
+	}
```
Comment (mminella) [inline spring-batch-infrastructure/src/test/java/org/springframework/batch/item/database/JpaItemWriterPersistTests.java]: This test doesn't validate anything unless I'm missing something...

**Comment by pivotal-issuemaster** [comment]
@Naros Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-batch&pullRequestId=399) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@Naros Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-batch&pullRequestId=399)!

**Comment by benas** [comment]
Rebased and merged as 229fed4e2468a7ae1f6775fdb7729955e4d4189e.

The newly added `JpaItemWriterPersistTests` is almost identical to the existing `JpaItemWriterTests` except the new test `testPersist`. So I moved this test to `JpaItemWriterTests` to avoid code duplication.

I also added an [integration test](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-infrastructure-tests/src/test/java/org/springframework/batch/item/database/JpaItemWriterIntegrationTests.java) for both `merge` and `persist` operations. This served as a base for performance comparison with 1M items. Here are the average results  of 5 executions on my laptop (Macbook Pro 16Go RAM, 2.9 GHz Intel Core i7 CPU, MacOS Mojave 10.14.5, Oracle JDK 1.8.0_201):

* merge: 19563ms
* persist: 9124ms

So in sum, using persist is 2 times faster than using merge. This is awesome because for a file ingestion job using JPA, this new option can make the job 2 times faster!

Thank you @Naros for your contribution! 👍

File: spring-batch-infrastructure/src/test/java/org/springframework/batch/item/database/JpaItemWriterPersistTests.java (hunk 1 above)
Comment (benas) [reply spring-batch-infrastructure/src/test/java/org/springframework/batch/item/database/JpaItemWriterPersistTests.java]: Fixed on merge (See 56ef0f35a9a5842906dbd5cb22555e28a023ee0f).

**Pull request closed without merging**